packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -523,30 +538,72 @@ export class KubernetesClient {
     }
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  getTags(tags: any[]): any[] {
+    for (const tag of tags) {
+      if (tag.tag === 'tag:yaml.org,2002:int') {
+        const newTag = { ...tag };
+        newTag.test = /^(0[0-7][0-7][0-7])$/;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        newTag.resolve = (str: any) => parseInt(str, 8);
+        tags.unshift(newTag);
+        break;
+      }
+    }
+    return tags;
+  }
+
+  // load yaml file and extract manifests
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  async loadManifestsFromFile(file: string): Promise<any[]> {
+    // throw exception if file does not exist
+    if (!fs.existsSync(file)) {
+      throw new Error(`File ${file} does not exist`);
+    }
+
+    // load file and create resources
+    const content = await fs.promises.readFile(file, 'utf-8');
+
+    const manifests = parseAllDocuments(content, { customTags: this.getTags });
+    return manifests.map(manifest => manifest.toJSON());
+  }
+
+  async createResourcesFromFile(context: string, filePath: string, namespace: string): Promise<void> {
+    const manifests = await this.loadManifestsFromFile(filePath);
+    return this.createResources(context, manifests, namespace);
+  }
+
   /**
    * Create Kubernetes resources on the specified cluster. Resources are create sequentially.
    *
    * @param context the context name to use
    * @param manifests the list of Kubernetes resources to create
    */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  async createResources(context: string, manifests: any[]): Promise<void> {
+  async createResources(context: string, manifests: any[], optionalNamespace?: string): Promise<void> {
```

Comment:
Running this twice in a row will unfortunately produce an error when replaying / reapplying a YAML. We should be using applyResources / equivalent for this functionality.

`kubectl create` vs `kubectl apply`.

For example, if we were to use the "deploy a YAML" functionality twice, it'll work fine with an "apply" but not with "create". As create will complain when ran a second time. 

This means that if we make changes to the YAML, instead we'll have to manually go through Kubernetes and delete the previous deployment before running "Deploy to Kubernetes" again. But if we were to use "apply" it'll work as it'll detect / compare changes and reapply them.